src/aspire/commands/apple.py

```diff
@@ -1,30 +1,130 @@
 
+logger = logging.getLogger(__name__)
+
 
 @click.command()
 @click.option(
     "--mrc_dir", help="Path to folder containing all mrc files for particle picking"
 )
 @click.option("--mrc_file", help="Path to a single mrc file for particle picking")
+@click.option(
+    "--create_jpg", is_flag=True, help="save *.jpg files for picked particles."
+)
 @click.option(
     "--output_dir",
     help="Path to folder to save *.star files. If unspecified, no star files are created.",
 )
 @click.option(
-    "--create_jpg", is_flag=True, help="save *.jpg files for picked particles."
+    "--particle_size",
+    help="Particle Size in pixels.  Many other options have internal defaults based off of Particle Size.",
+    required=True,
+    type=int,
 )
-def apple(mrc_dir, mrc_file, output_dir, create_jpg):
+@click.option("--max_particle_size", help="", default=None, type=int)
+@click.option("--min_particle_size", help="", default=None, type=int)
+@click.option("--minimum_overlap_amount", help="", default=None, type=int)
+@click.option("--query_image_size", help="", default=52)
+@click.option("--tau1", help="", default=None, type=int)
+@click.option("--tau2", help="", default=None, type=int)
+@click.option("--container_size", help="", default=450)
+@click.option("--model", help="", default="svm")
+@click.option(
+    "--model_opts",
+    help="Optional JSON dictionary representing specific options corresponding to `model`.",
+    default=None,
+    type=str,
+)
+@click.option("--mrc_margin_left", help="", default=99)
+@click.option("--mrc_margin_right", help="", default=100)
+@click.option("--mrc_margin_top", help="", default=99)
+@click.option("--mrc_margin_bottom", help="", default=100)
+@click.option("--mrc_shrink_factor", help="", default=2)
+@click.option("--mrc_gauss_filter_size", help="", default=15)
+@click.option("--mrc_gauss_filter_sigma", help="", default=0.5)
+@click.option("--response_thresh_norm_factor", help="", default=20)
+@click.option("--conv_map_nthreads", help="", default=4)
+@click.option(
+    "--n_processes",
+    help="Concurrent processes to spawn."
+    "May improve performance on very large machines."
+    "Otherwise use default.",
+    default=1,
+)
+def apple(
+    mrc_dir,
+    mrc_file,
+    create_jpg,
+    output_dir,
+    particle_size,
+    max_particle_size,
+    min_particle_size,
+    query_image_size,
+    tau1,
+    tau2,
+    minimum_overlap_amount,
+    container_size,
+    model,
+    model_opts,
+    mrc_margin_left,
+    mrc_margin_right,
+    mrc_margin_top,
+    mrc_margin_bottom,
+    mrc_shrink_factor,
+    mrc_gauss_filter_size,
+    mrc_gauss_filter_sigma,
+    response_thresh_norm_factor,
+    conv_map_nthreads,
+    n_processes,
+):
     """Pick and save particles from one or more mrc files."""
 
     # Exactly one of mrc_dir/mrc_file should be specified.
     # We handle this manually here until Click supports mutually exclusive options.
     if all([mrc_dir, mrc_file]) or not any([mrc_dir, mrc_file]):
```

Comment:
I poked around a bit and found https://github.com/click-contrib/click-option-group, but it seems like overkill for this situation. But maybe useful in case our command line flags grow more complicated.

Reply:
Right. Yes, more elaborate click configuration has come up before, but I'm not sure we want to go down that road yet.

In this case we can probably just take an `mrc_path` that is either a file or a directory. I'll see what that looks like and maybe push a commit for it.